src/shell/layout/floating/mod.rs

```diff
@@ -110,9 +110,12 @@ impl FloatingLayout {
                 mapped.min_size().unwrap_or((0, 0).into()),
                 mapped.max_size().unwrap_or((0, 0).into()),
             );
-            if win_geo.size.w > geometry.size.w / 3 * 2 {
+            if win_geo.size.w > max_size.w
+                || win_geo.size.w < min_size.w
+                || win_geo.size.w > geometry.size.w
+            {
```

Comment:
I don't think this code is the issue here, given it is supposed to be applied for freshly created windows, where we need to find a new fitting size. And that mostly works fine. What problematic here, is that with `last_geometry` set we should just always use that. So there is basically an `else` missing in lines 107/108.